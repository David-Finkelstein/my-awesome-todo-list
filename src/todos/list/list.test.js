import React from 'react';
import { mount } from "enzyme";
import immutable from "immutable";
import { mountToJson } from "enzyme-to-json";
import { Provider } from "react-redux";
import {combineReducers, createStore} from "redux";
import { BrowserRouter as Router, Route } from "react-router-dom";

import List from './index'

const reducer = combineReducers({ todoList: () => { return immutable.fromJS({}) } });
const store = createStore(reducer, undefined, undefined);

describe('<List />', () => {
    it('takes a snapshot of the component and check if the prev snapshot equals to the current', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router>
                    <Route exact path="/" component={List}/>
                </Router>
            </Provider>
        );

        expect(mountToJson(wrapper)).toMatchSnapshot();
    });
});