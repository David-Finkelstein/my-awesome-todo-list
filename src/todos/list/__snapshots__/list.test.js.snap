// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> takes a snapshot of the component and check if the prev snapshot equals to the current 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      >
        <Connect(List)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <List
            deleteTodo={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            todoArray={Array []}
            toggleTodoStatus={[Function]}
          >
            <ul
              className="list-unstyled"
            />
          </List>
        </Connect(List)>
      </Route>
    </Router>
  </BrowserRouter>
</Provider>
`;
